core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/MarecoAllowance.java

```diff
@@ -0,0 +1,294 @@
+                count++)  {
+            x = x - newtonsMethod(x, maxSpeed) / newtonsMethodDerivative(x);  //Newtons method.
+        }
+
+        if (abs(newtonsMethod(x, maxSpeed)) <= tolerance) {
+            return x * DICHOTOMY_MARGIN;
+        } else {
+            return maxSpeed * DICHOTOMY_MARGIN; // if no value has been found return a high value to have some margin
+        }
+    }
+
+    @Override
+    protected double getFirstGuess(Envelope envelope) {
+        return (this.getFirstHighEstimate(envelope) + this.getFirstLowEstimate()) / 2;
+    }
+
+    /** Finds the end position of the coasting phases, that will then be generated backwards
+     * with generateCoastingSpeedControllerAtPosition()
+     * Coasting phases can be generated for two different reasons
+     * - a deceleration phase where braking will be replaced by coasting
+     * - before an accelerating slope */
+    private List<Double> findEndOfCoastingPositions(Envelope envelope, double v1) {
+        var res = new ArrayList<Double>();
+        double vf = vf(v1);
+
+        // coasting before deceleration phases
+        var brakingEnvelopeParts = findBrakingEnvelopeParts(envelope);
+        for (var part : brakingEnvelopeParts) {
+            double targetSpeed = part.getEndSpeed();
+            // if that LimitAnnounceSpeedController is above v1 that means it will not have an impact here
+            if (targetSpeed > v1)
+                continue;
+            // deceleration phases that are entirely above vf
+            if (targetSpeed > vf) {
+                res.add(part.getEndPos());
+                continue;
+            }
+            // deceleration phases that cross vf
+            if (part.getMaxSpeed() > vf)
+                res.add(part.interpolatePosition(vf));
+        }
+
+        // coasting before accelerating slopes
+        var acceleratingSlopes =
+                findAcceleratingSlopes(envelope, rollingStock, vf);
+        double wle = rollingStock.getRollingResistance(v1) * v1 * vf / (v1 - vf);
+        for (var slope : acceleratingSlopes) {
+            // formulas given my MARECO
+            // giving the optimized speed v the train should have when entering the accelerating slope
+            // this speed v might not be reached if the slope is not long enough, then we just enter the slope with
+            // the lowest possible speed that will catch up with target speed at the end
+            double v = 1 / (rollingStock.getRollingResistance(v1)
+                    / (wle * (1 - slope.previousAcceleration / slope.acceleration)) + 1 / v1);
+            double target = slope.targetSpeed;
+            double requiredAcceleratingDistance = Math.max((target * target - v * v) / (2 * slope.acceleration), 0);
+            double positionWhereTargetSpeedIsReached = slope.beginPosition + requiredAcceleratingDistance;
+            res.add(Math.min(slope.endPosition, positionWhereTargetSpeedIsReached));
```

Comment:
We don't seem to have any test with accelerating slope, it may be a problem

Reply:
correct, I've opened #484 (I'd rather test it after this is merged)